engine/src/main/java/org/terasology/world/chunks/localChunkProvider/LocalChunkProvider.java

```diff
@@ -338,15 +245,9 @@ private void generateQueuedEntities(EntityStore store) {
 
     @Override
     public void beginUpdate() {
-        regionLock.readLock().lock();
-        try {
-            updateRelevance();
-            deactivateBlocks();
-            checkForUnload();
-            makeChunksAvailable();
-        } finally {
-            regionLock.readLock().unlock();
-        }
```

Comment:
is this update all done on the main thread. I was looking around and It seems to be single threaded?

Reply:
regionLock really need used only by `updateRelevance`, which was moved to RelevanceSystem(regionLock moved there also)